cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -356,14 +360,38 @@ public void error(Throwable cause) {
     }
   }
 
+  /**
+   * On an authorization-enabled cluster, if caching is enabled too, updates the cache in the
+   * {@link AuthorizationEnforcementService} with the privileges of the user running the program.
+   */
+  private void updateAuthCache() {
```

Comment:
Why this method belongs to this class? Shouldn't it be part of what the `AuthorizationEnforcementService` do as part of it's startup?